README.md

```diff
@@ -28,102 +28,45 @@ composer require softonic/laravel-transactional-event-publisher
 ```bash
 php artisan vendor:publish --provider="Softonic\TransactionalEventPublisher\ServiceProvider" --tag=migrations
 ```
-and execute the migrations
+and execute them
 ```bash
 php artisan migrate
 ```
 
-### Registering Models
-
-To choose what models should send domain events, you need to attach the `\Softonic\TransactionalEventPublisher\ModelObserver` observer class.
-
-Example:
-
-```
-...
-
-use App\Post as MyModel;
-
-class AppServiceProvider extends ServiceProvider
-{
-    public function boot()
-    {
-        MyModel::observe(\Softonic\TransactionalEventPublisher\Observers\ModelObserver::class);
-    }
-    ...
-}
-```
-
-### Custom middlewares
-
-The middlewares should implement the `Softonic\TransactionalEventPublisher\Contracts\EventStoreMiddlewareContract` interface.
-Its purpose is store the domain event provided, so you can implement any storage for domain events.
-
-### Custom messages
-
-The `transactional-event.message` class must implements `EventMessageContract` and `transactional-event.middleware` class must implements `EventStoreMiddlewareContract`
+#### Amqp middleware
 
-### Sending all the events stored in database
+This middleware publishes the events to an AMQP system. You just need to configure the AMQP connection using the configuration file or environmental variables.
+As you can see, in the configuration you won't be able to define a queue. This is because the library just publishes the message to an exchange and is the events collector responsibility to declare the needed queues with the needed bindings.
 
-Sometimes you will need to send all events stored in the database. To do this, you can use the command `php artisan tinker event-sourcing:emit-all`.
+### Publishing events in batches to improve performance
 
-The command allows you to use a specific queue connection to emit all the events and a specific database connection with unbuffered connection.
-These parameters are optional but allows you to send several millions of events in a short time without consume a lot of memory.
+We provide a command to continuously publish events in batches in real time.
+You can find its signature in `Softonic\TransactionalEventPublisher\Console\Commands\EmitEvents`.
+It will publish the events in batches of 100 by default if the option is not set.
+You just need to create a job that will run indefinitely with the command `php artisan event-sourcing:emit`.
 
-The `queueConnection` argument lets you choose a high performance queue driver like redis.
-The `--unbufferedConnection` options allows you to use [unbuffered queries](https://dev.mysql.com/doc/apis-php/en/apis-php-mysqlinfo.concepts.buffering.html) in MySQL to retrieve a large amount of rows without consuming all the memory.
+#### Sending all the events stored in database
 
+Sometimes you will need to send all the events stored in the database. To do it, you can run the previous command with the option `--allEvents`.
```

Comment:
What happens when all events are emitted? Are they emitted with the original timestamp, a new one? What happens when they are emitted as usual, if they are throttled they are emitted with a new timestamp?

Reply:
With the original one, it's always with the EventMessage creation time. We're emitting them with the new timestamp in a customitzation we have in our projects only, but not in this public library because of the way we consume the events.